### PR TITLE
 Add test case for checking all workflows correct description

### PR DESCRIPTION
**Pull request by ShatalovYaroslav**

### CODE REVIEW

File: src/test/java/org/ow2/proactive/ProActiveExamplesValidityTest.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+    @Parameterized.Parameters(name = "{index}: testing workflow - {0}")
+    public static Collection<String> data() {
+        Filter<Path> directoryFilter = new Filter<Path>() {
+            @Override
+            public boolean accept(Path entry) throws IOException {
+                return Files.isDirectory(entry) && Files.exists(Paths.get(entry.toString(), "METADATA.json"));
+            }
+        };
+
```
Comment (Mykhailenko) [inline src/test/java/org/ow2/proactive/ProActiveExamplesValidityTest.java]: Maybe you can use Lambda hare.

File: src/test/java/org/ow2/proactive/ProActiveExamplesValidityTest.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        List<String> files = new ArrayList<>();
+        if (Files.isDirectory(Paths.get(""))) {
+            try (DirectoryStream<Path> directoryStream = Files.newDirectoryStream(Paths.get(""), directoryFilter)) {
+                directoryStream.forEach(directoryPath -> {
+                    Path workflowDirectory = Paths.get(directoryPath.toString(), PATH_TO_WORKFLOWS);
+                    try (DirectoryStream<Path> stream = Files.newDirectoryStream(workflowDirectory, "*.xml")) {
+                        stream.forEach(workflowPath -> files.add(workflowPath.toString()));
+                    } catch (IOException e) {
+                        throw new RuntimeException(e);
+                    }
```
Comment (Mykhailenko) [inline src/test/java/org/ow2/proactive/ProActiveExamplesValidityTest.java]: Are you sure you need to wrap exception?

File: src/test/java/org/ow2/proactive/ProActiveExamplesValidityTest.java [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+                });
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
```
Comment (Mykhailenko) [inline src/test/java/org/ow2/proactive/ProActiveExamplesValidityTest.java]: same comment

File: src/test/java/org/ow2/proactive/ProActiveExamplesValidityTest.java [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+    @Test
+    public void testWfDescription() throws IOException, TransformerException, ParserConfigurationException {
+        JobFactory factory = JobFactory.getFactory();
+
+        TaskFlowJob job = null;
+        try {
+            job = (TaskFlowJob) factory.createJob(filePath);
+        } catch (JobCreationException e) {
+            throw new RuntimeException("The workflow is not valid", e);
+        }
```
Comment (Mykhailenko) [inline src/test/java/org/ow2/proactive/ProActiveExamplesValidityTest.java]: same

File: src/test/java/org/ow2/proactive/ProActiveExamplesValidityTest.java [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+                                            Files.exists(Paths.get(entry.toString(), "METADATA.json")))
+                           .map(directoryPath -> Paths.get(directoryPath.toString(), PATH_TO_WORKFLOWS))
+                           .flatMap(resourcesPath -> {
+                               try {
+                                   return Files.list(resourcesPath).filter(file -> file.toString().endsWith(".xml"));
+                               } catch (IOException e) {
+                                   throw new RuntimeException(e);
+                               }
+                           })
+                           .forEach(workflowPath -> files.add(workflowPath.toString()));
```
Comment (Mykhailenko) [inline src/test/java/org/ow2/proactive/ProActiveExamplesValidityTest.java]: instead of forEash you can use map and collect

**Comment by ShatalovYaroslav** [comment]
retest this please

**Comment by ShatalovYaroslav** [comment]
retest this please

**Pull request merged**